test/issues.jl

```diff
@@ -155,6 +155,79 @@ try
         end
     end
 
+    @testset "issue 515" begin
+        # https://github.com/JuliaCloud/AWS.jl/issues/515
+        n = 100
+        n_incomplete = n - 1
+        data = rand(UInt8, n)
+
+        @testset "Fail 2 attempts then succeed" begin
+            i = 2
+            ATTEMPT_NUM = 0 # reset
+            patch = if AWS.DEFAULT_BACKEND[] isa AWS.HTTPBackend
+                @patch function HTTP.request(args...; response_stream, kwargs...)
+                    ATTEMPT_NUM += 1
+                    if ATTEMPT_NUM <= i
+                        write(response_stream, data[1:n_incomplete]) # an incomplete stream that shouldn't be retained
+                        throw(HTTP.IOError(EOFError(), "msg"))
+                    else
+                        write(response_stream, data)
+                        return HTTP.Response(200, "{\"Location\": \"us-east-1\"}")
+                    end
+                end
+            elseif AWS.DEFAULT_BACKEND[] isa AWS.DownloadsBackend
+                @patch function Downloads.request(args...; output, kwargs...)
+                    ATTEMPT_NUM += 1
+                    if ATTEMPT_NUM <= i
+                        write(output, data[1:n_incomplete]) # an incomplete stream that shouldn't be retained
+                        throw(Downloads.RequestError("", 18, "transfer closed with 1 bytes remaining to read", Downloads.Response("http", "", 200, "HTTP/1.1 200 OK", ["content-length" => "64"])))
```

Comment:
I feel like I'm missing something.
```suggestion
                        throw(Downloads.RequestError("", 18, "transfer closed with 1 bytes remaining to read", Downloads.Response("http", "", 200, "HTTP/1.1 200 OK", ["content-length" => string(n)])))
```